torch/return_types.py

```diff
@@ -15,8 +15,13 @@ def structseq_unflatten(values, context):
 
     torch.utils._pytree._register_pytree_node(cls, structseq_flatten, structseq_unflatten)
 
+DONT_EXPORT = {
+    "_linalg_solve",
+    "_linalg_solve_out",
+}
+
 for name in dir(return_types):
-    if name.startswith('__'):
+    if name.startswith('__') or name in DONT_EXPORT:
```

Comment:
@anjali411 @albanD I was getting the following error in `test_correct_module_names`:
### torch.return_types._linalg_solve:
  - Is public: it is inside the module's (`torch.return_types`) `__all__`
  - Does NOT look public: because it starts with `_` (`_linalg_solve`)
  - You can do either of these two things to fix this problem:
    - To make it NOT public: remove it from the modules's (`torch.return_types`) `__all__`
    - To make it look public: remove the `_` at the beginning of the name
### torch.return_types._linalg_solve_out:
  - Is public: it is inside the module's (`torch.return_types`) `__all__`
  - Does NOT look public: because it starts with `_` (`_linalg_solve_out`)
  - You can do either of these two things to fix this problem:
    - To make it NOT public: remove it from the modules's (`torch.return_types`) `__all__`
    - To make it look public: remove the `_` at the beginning of the name

could you confirm whether this fix is the expected fix, or how should this be approached?

Reply:
This is certainly not the right way to go: https://github.com/pytorch/pytorch/runs/6489328515?check_suite_focus=true
What should I do here? All I want is an auxiliary function that's not exposed in the public API, but I want to be able to implement derivatives for it and all that.

Reply:
The test seems to still find them in `__all__`?

Reply:
No, the test that fails now is a different test (it's the test that uses the namespace `return_types`.
Note that this is just an implementation detail (I believe) of functions that have multiple returns.
Perhaps the `return_types` namespace should be handled differently in `test_correct_module_names`? Again, not sure what that namespace is used for really.

Reply:
Fixed (they should not be added to `__all__` but they should be added to globals).